extensions/medplum/actions/createCalculationObservation/createCalculationObservation.ts

```diff
@@ -19,82 +23,85 @@ export const createCalculationObservation: Action<
+    const awellSdk = await helpers.awellSdk()
 
-    const awellSdk = new AwellSdk({
-      apiUrl: awellApiUrl ?? '',
-      apiKey: awellApiKey ?? '',
-    })
-
-    const activities = await awellSdk.getPathwayActivities({
-      pathway_id: pathway.id,
-    })
-
-    const currentActivity = activities.find((a) => a.id === activity.id)
+    let latestCalculationActivity: Activity
 
-    if (isNil(currentActivity))
-      throw new Error('Cannot find the current activity')
-
-    const currentStepId = currentActivity.context?.step_id
-
-    if (isNil(currentStepId))
-      throw new Error('Could not find step ID of the current activity')
-
-    const activitiesInCurrentStep = activities.filter(
-      (a) =>
-        a.context?.step_id === currentStepId &&
-        a.status === 'DONE' &&
-        a.date <= currentActivity.date
-    )
+    try {
+      const res = await getLastCalculationActivityInCurrentStep({
+        awellSdk,
+        pathwayId: pathway.id,
+        currentActivityId: activity.id,
+      })
 
-    // Grab the most recent calculation activity
-    const calculationActivityToPushToMedplum = activitiesInCurrentStep.filter(
-      (a) => a.object.type === 'CALCULATION'
-    )[0]
+      if (isNil(res)) {
+        throw new Error('No calculation activity found in the current step')
+      }
 
-    if (isNil(calculationActivityToPushToMedplum))
-      throw new Error(
-        'No calculation action found in the current step to push to Medplum'
-      )
+      latestCalculationActivity = res
+    } catch (error) {
+      const err = error as Error
+      await onError({
+        events: [
+          addActivityEventLog({
+            message: err.message,
+          }),
+        ],
+      })
+      return
+    }
 
-    const calculationResults = await awellSdk.getCalculationResults({
-      pathway_id: pathway.id,
-      activity_id: calculationActivityToPushToMedplum.id,
+    const calculationResultsResponse = await awellSdk.orchestration.query({
+      calculationResults: {
+        __args: {
+          pathway_id: pathway.id,
+          activity_id: latestCalculationActivity.id,
+        },
+        success: true,
+        result: {
+          __scalar: true,
+        },
+      },
     })
 
     const derivedFrom: Reference<QuestionnaireResponse> | null = !isEmpty(
-      input.questionnaireResponseId
+      input.questionnaireResponseId,
     )
       ? {
           reference: `QuestionnaireResponse/${
             extractResourceId(
               input.questionnaireResponseId ?? '',
-              'QuestionnaireResponse'
+              'QuestionnaireResponse',
             ) ?? 'undefined'
           }`,
         }
       : null
 
+    /**
+     * We currently only support one result per calculation activity so
+     * we use an imperfect heuristic to just get the first result of the calculation
+     */
+    const score =
+      calculationResultsResponse.calculationResults?.result[0]?.value
+    const isValidScore =
```

Comment:
can't we check for the missing status aswell here? I believe we retrieve it.
also, the plan is to keep supporting just one result?

Reply:
Yes, we could check for the missing status as well. However, the result.status field in the GraphQL type is defined as a plain string, not an enum, which would have made the reference simpler. To handle this, I opted to perform the check directly against the string value.

As for the current functionality, the action only supports numeric results, so this stricter check aligns with that constraint.